assets/tipuesearch/tipuesearch_content.js

```diff
@@ -4,88 +4,89 @@
+    {% for page in site.html_pages %}
+      {% unless page.exclude_from_search == true or excluded_files contains page.path %}
+        {% assign has_excluded_taxonomy = false %}
+        {% for tag in page.tags %}
+          {% if excluded_taxonomies contains tag %}
+            {% assign has_excluded_taxonomy = true %}
+          {% endif %}
+        {% endfor %}
+        {% for category in page.categories %}
+          {% if excluded_taxonomies contains category %}
+            {% assign has_excluded_taxonomy = true %}
+          {% endif %}
+        {% endfor %}
+        {% unless has_excluded_taxonomy == true %}
+          {% assign index = index | push: page | uniq %}
+        {% endunless %}
+      {% endunless %}
+    {% endfor %}
+  {% endif %}
+  {% for collection in site.tipue_search.include.collections %}
+    {% assign documents = site.documents | where:"collection",collection %}
+    {% for document in documents %}
+      {% unless document.exclude_from_search == true or excluded_files contains document.path %}
+        {% assign has_excluded_taxonomy = false %}
+        {% for tag in document.tags %}
+          {% if excluded_taxonomies contains tag %}
+            {% assign has_excluded_taxonomy = true %}
+          {% endif %}
+        {% endfor %}
+        {% for category in document.categories %}
+          {% if excluded_taxonomies contains category %}
+            {% assign has_excluded_taxonomy = true %}
+          {% endif %}
+        {% endfor %}
+        {% unless has_excluded_taxonomy == true %}
+          {% assign index = index | push: document | uniq %}
+        {% endunless %}
+      {% endunless %}
+    {% endfor %}
   {% endfor %}
-  {% for category in document.categories %}
-    {% assign taxonomies = taxonomies | push: category | uniq %}
+  {% for document in index %}
+    {% assign taxonomies = "" | split: "" %}
+    {% for tag in document.tags %}
+      {% assign taxonomies = taxonomies | push: tag | uniq %}
+    {% endfor %}
+    {% for category in document.categories %}
+      {% assign taxonomies = taxonomies | push: category | uniq %}
+    {% endfor %}
+    {
+      "title": {{ document.title | smartify | strip_html | normalize_whitespace | jsonify }},
+      "text": {{ document.content | markdownify | strip_html | normalize_whitespace | jsonify }},
+      "tags": {{ taxonomies | join: " " | normalize_whitespace | jsonify }},
+      "url": {{ document.url | relative_url | jsonify }}
+    }{% unless forloop.last %},{% endunless %}
   {% endfor %}
-  {
-    "title": {{ document.title | smartify | strip_html | normalize_whitespace | jsonify }},
-    "text": {{ document.content | strip_html | normalize_whitespace | jsonify }},
-    "tags": {{ taxonomies | join: " " | normalize_whitespace | jsonify }},
-    "url": {{ document.url | relative_url | jsonify }}
-  }{% unless forloop.last %},{% endunless %}
-{% endfor %}
-]};
+{% endcapture %}
+var tipuesearch = {"pages": [ {{ pages | normalize_whitespace }} ]};
```

Comment:
Good idea however I'd rather not use a capture block or `normalize_whitespace`. I will be adding whitespace control when Jekyll 3.5 is update on Github pages: https://github.com/jekylltools/jekyll-tipue-search/issues/3

Here is what `tipuesearch_content.js` will look like after using whitespace control: https://gist.github.com/jekylltools/6931f0d02b7edececdb258450fe2d917